### PR TITLE
Feat/BNGP-3611-choose-if-acting-on-behalf-of-client-or-not

### PR DESCRIPTION
**Pull request by KiranVarma02**

### CODE REVIEW

File: packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  describe('POST', () => {
+    it('Should continue journey to APPLICANT_DETAILS_CONFIRM if yes is chosen and remove 1 legal party', async () => {
+      const request = {
+        yar: redisMap,
+        payload: { isApplicantAgent: 'yes' }
+      }
+
+      await isApplicantAgent.default[1].handler(request, h)
+
+      expect(viewResult).toEqual(constants.routes.APPLICANT_DETAILS_IS_AGENT)
```
Comment (pwadmore-ea) [inline packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js:55]: Add TODO comment to refactor when required routing is available.

File: packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    it('Should continue journey to APPLICANT_ROLE_AS_LANDOWNER if no is chosen', async () => {
+      const request = {
+        yar: redisMap,
+        payload: { isApplicantAgent: 'no' }
+      }
+
+      await isApplicantAgent.default[1].handler(request, h)
+
+      expect(viewResult).toEqual(constants.routes.APPLICANT_DETAILS_IS_AGENT)
```
Comment (pwadmore-ea) [inline packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js:66]: Add TODO comment to refactor when required routing is available.

File: packages/webapp/src/routes/__tests__/land/register-land-task-list.spec.js [hunk 3]
```diff
@@ -38,8 +38,22 @@ describe(url, () => {
...
+          // TODO: remove skip and test should pass after all routes are developed
+          // expect(contextResult.registrationTasks.taskList[0]).toEqual({
+          //   taskTitle: 'Applicant information',
+          //   tasks: [
+          //     {
+          //       title: 'Add details about the person applying',
+          //       status: 'NOT STARTED',
+          //       completedTaskUrl: '/land/check-ownership-details',
+          //       startTaskUrl: '/land/applicant-details-is-agent',
+          //       inProgressUrl: '',
```
Comment (pwadmore-ea) [inline packages/webapp/src/routes/__tests__/land/register-land-task-list.spec.js:51]: Refactor to match inProgressUrl used in the application configuration.

File: packages/webapp/src/utils/register-task-list.js [hunk 4]
```diff
@@ -2,6 +2,19 @@ import constants from './constants.js'
 
 const registerTaskList = {
   taskList: [
+    {
+      taskTitle: 'Applicant information',
+      tasks: [
+        {
+          title: 'Add details about the person applying',
+          status: constants.DEFAULT_REGISTRATION_TASK_STATUS,
+          completedTaskUrl: constants.routes.CHECK_OWNERSHIP_DETAILS,
```
Comment (pwadmore-ea) [inline packages/webapp/src/utils/register-task-list.js:11]: This does not look correct.

File: packages/webapp/src/__mock-data__/test-application.js [hunk 5]
```diff
@@ -2,6 +2,19 @@ const dataString = `
...
+      {
+        "taskTitle": "Applicant information",
+        "tasks": [
+          {
+            "title": "Add details about the person applying",
+            "status": "COMPLETED",
+            "completedTaskUrl": "/land/applicant-details-is-agent",
+            "startTaskUrl": "/land/applicant-details-is-agent",
+            "inProgressUrl": "/land/add-landowners",
+            "id": "applicant-details-is-agent"
```
Comment (pwadmore-ea) [inline packages/webapp/src/__mock-data__/test-application.js:14]: This does not match the id in [packages/webapp/src/utils/register-task-list.js](https://github.com/DEFRA/biodiversity-net-gain-service/pull/482/files#diff-a52c8b94614c8e2ba7fdd4c53843fe7cdedf2d333160e04eba4aefe6f73f8bb1)

File: packages/webapp/src/routes/__tests__/land/register-land-task-list.spec.js [hunk 6]
```diff
@@ -38,8 +38,22 @@ describe(url, () => {
...
-          expect(contextResult.registrationTasks.taskList[0]).toEqual({
+          expect(contextResult.registrationTasks.taskList.length).toEqual(5)
+          // TODO: remove skip and test should pass after all routes are developed
+          // expect(contextResult.registrationTasks.taskList[0]).toEqual({
+          //   taskTitle: 'Applicant information',
+          //   tasks: [
+          //     {
+          //       title: 'Add details about the person applying',
+          //       status: 'NOT STARTED',
+          //       completedTaskUrl: '/land/check-ownership-details',
```
Comment (pwadmore-ea) [inline packages/webapp/src/routes/__tests__/land/register-land-task-list.spec.js:49]: This does not look correct

**Review by pwadmore-ea — CHANGES_REQUESTED** [review]

**Comment by pwadmore-ea** [comment]
@KiranVarma02  - There are a couple of code smells to be resolved.

File: packages/webapp/src/__mock-data__/test-application.js [hunk 7]
```diff
@@ -2,6 +2,19 @@ const dataString = `
...
     "taskList": [
+      {
+        "taskTitle": "Applicant information",
+        "tasks": [
+          {
+            "title": "Add details about the person applying",
+            "status": "COMPLETED",
+            "completedTaskUrl": "/land/applicant-details-is-agent",
+            "startTaskUrl": "/land/applicant-details-is-agent",
+            "inProgressUrl": "/land/add-landowners",
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply packages/webapp/src/__mock-data__/test-application.js:13]: completed should be different but I don't know what that is at the moment. I will put TODO

File: packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js (hunk 2 above)
Comment (KiranVarma02) [reply packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js:66]: I guess your comment is ment for no test above. This test when there is no answer will work now.

File: packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+  describe('GET', () => {
+    it(`should render the ${url.substring(1)} view`, async () => {
+      await submitGetRequest({ url })
+    })
+
+    it('should show correct party to be remove', async () => {
+      const request = {
+        yar: redisMap,
+        query: { orgId: '0' }
+      }
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js:38]: done

File: packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+      await isApplicantAgent.default[0].handler(request, h)
+
+      expect(viewResult).toEqual(constants.views.APPLICANT_DETAILS_IS_AGENT)
+    })
+  })
+
+  describe('POST', () => {
+    it('Should continue journey to APPLICANT_DETAILS_CONFIRM if yes is chosen and remove 1 legal party', async () => {
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js:47]: done

File: packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js (hunk 1 above)
Comment (KiranVarma02) [reply packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js:55]: done

File: packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js (hunk 2 above)
Comment (KiranVarma02) [reply packages/webapp/src/routes/__tests__/land/applicant-details-is-agent.spec.js:66]: done

File: packages/webapp/src/routes/__tests__/land/register-land-task-list.spec.js (hunk 3 above)
Comment (KiranVarma02) [reply packages/webapp/src/routes/__tests__/land/register-land-task-list.spec.js:51]: TODO added to fix once all routes are complete

**Review by pwadmore-ea — APPROVED** [review]

**Pull request merged**